src/samples/audio/smart_amp_test_ipc4.c

```diff
@@ -358,8 +349,32 @@ static const struct module_interface smart_amp_test_interface = {
 	.free = smart_amp_free
 };
 
+#ifndef __MODULE_BUILD__
```

Comment:
please have a look at https://github.com/thesofproject/sof/pull/8180/commits/03aad36103cc93c17b1fb32f76611910a0188a22 and the first hunk in https://github.com/thesofproject/sof/pull/8180/commits/9e6dd7ec71d34de8c8fe722ba6de0b5333adafda

Reply:
OK, we need to avoid common names to avoid collision with Zephyr, can we rename to ```__SOF_MODULE_SERVICE_BUILD__``` and this would indicate a SOF macro for modules using the services API. Not a blocker for this PR, can be incrementally updated.

Reply:
sure, this is for a demo. We still need a lot of work to include loadable module building system